hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/JmsSourceBuilder.java

```diff
@@ -46,114 +48,171 @@
         this.username = username;
         this.password = password;
         return this;
     }
 
     /**
-     * Sets the function which creates the connection from connection factory.
+     * Sets the function which creates the connection using the connection
+     * factory.
      * <p>
-     * If not provided, the builder creates a function which uses {@code
-     * ConnectionFactory#createConnection(username, password)} to create the
-     * connection. See {@link #connectionParams(String, String)}.
+     * If not provided, this function is used:
+     * <pre>
+     *     connectionFn = factory -> factory instanceof XAConnectionFactory
+     *            ? ((XAConnectionFactory) factory).createXAConnection(username, password)
+     *            : factory.createConnection(username, password);
+     * </pre>
+     * That means it creates an XA connection if the factory is an XA factory.
+     * The user name and password set with {@link #connectionParams} are used.
+     *
+     * @return this instance for fluent API
      */
+    @Nonnull
     public JmsSourceBuilder connectionFn(
-            @Nonnull FunctionEx<? super ConnectionFactory, ? extends Connection> connectionFn
+            @Nullable FunctionEx<? super ConnectionFactory, ? extends Connection> connectionFn
     ) {
         checkSerializable(connectionFn, "connectionFn");
         this.connectionFn = connectionFn;
         return this;
     }
 
     /**
-     * Sets the session parameters. If {@code sessionFn} is provided these
-     * parameters are ignored.
+     * Sets the name of the destination (name of the topic or queue). If {@link
+     * #consumerFn(FunctionEx)} is provided, this parameter is ignored.
      *
-     * @param transacted       if true marks the session as transacted false otherwise,
-     *                         Default value is false.
-     * @param acknowledgeMode  sets the acknowledge mode of the session,
-     *                         Default value is {@code Session.AUTO_ACKNOWLEDGE}
+     * @return this instance for fluent API
      */
-    public JmsSourceBuilder sessionParams(boolean transacted, int acknowledgeMode) {
-        this.transacted = transacted;
-        this.acknowledgeMode = acknowledgeMode;
+    @Nonnull
+    public JmsSourceBuilder destinationName(@Nullable String destinationName) {
+        this.destinationName = destinationName;
         return this;
     }
 
     /**
-     * Sets the function which creates the session from connection.
-     * <p>
-     * If not provided, the builder creates a function which uses {@code
-     * Connection#createSession(boolean transacted, int acknowledgeMode)} to
-     * create the session. See {@link #sessionParams(boolean, int)}.
+     * Sets the function which creates the message consumer from session.
+     *
+     * <p>If not provided, {@code Session#createConsumer(destinationName)} is used
```

Comment:
I would strongly prefer you not introducing this style of using `<p>`, which is inconsistent with the rest of the codebase. It also makes it harder for me to spot any missing `<p>`s, a problem which becomes apparent only in the generated Javadoc.

Reply:
On the other hand it improves source code readability when you're _not_ interested in checking for missing `<p>`. And, most importantly, there's tool support for this style: I use "Wrap to Column" plugin and it doesn't support the other style without manually selecting a block to reformat. I tweaked the text many many times and I hate to reflow manually or being forced to select it before reformatting.

And I already sneaked quite a few instances into the code base :)

Reply:
I think we don't terminate <p> typically, but it should be on a separate line, not part of the paragraph.